src/main/kotlin/notification/domain/Notification.kt

```diff
@@ -2,4 +2,4 @@ package notification.domain
 
 import shared.domain.Identifier
 
-class Notification(val id: Identifier)
+class Notification(val id: Identifier, val noteId: Identifier)
```

Comment:
Why would you want an identifier for the notification?

A notification only has a note and a timestamp identifier that identifies them, if the noteId and the time is the same, that means that this notification already existed on the repository.

Plus, why we need a get and getAll methods? Only one get method is needed (one that returns a Collection<Notification> associated with this note. If the note doesn't have any notifications the collection will be empty). (Anyway, the "get" name for the method isn't the more convenient, see the discussion #43 and read the "domain language" rules, but be aware that as stated in #43, you should only rename and refactor things from THIS feature, not from any other).